index.js

```diff
@@ -356,6 +356,31 @@
     ($.Undefined  = type0('Undefined'))
   ];
 
+  //  ValidNumber :: Type
+  $.ValidNumber = $.NullaryType(
+    'sanctuary-def/ValidNumber',
+    function(x) { return $.Number.test(x) && !isNaN(x); }
+  );
+
+  //  FiniteNumber :: Type
+  $.FiniteNumber = $.NullaryType(
+    'sanctuary-def/FiniteNumber',
+    function(x) { return $.ValidNumber.test(x) && x !== Infinity &&
+                                                  x !== -Infinity; }
```

Comment:
We could use `isFinite(x)` in place of the second and third tests. This is actually important as `new Number(Infinity) !== Infinity`.